sources/packages/api/src/route-controllers/application/application.controller.ts

```diff
@@ -394,14 +408,28 @@ export class ApplicationController extends BaseController {
       );
     }
 
-    const updateResult = await this.applicationService.studentConfirmAssessment(
-      applicationId,
-      student.id,
-    );
-    if (updateResult.affected === 0) {
-      throw new UnprocessableEntityException(
-        `Confirmation of Assessment for the application id ${applicationId} failed.`,
+    // TODO: temporary code to allow the assessment to be retrieved using the application id.
+    const [originalAssessment] =
+      await this.assessmentService.getAssessmentsByApplicationId(
+        applicationId,
+        AssessmentTriggerType.OriginalAssessment,
       );
+    // TODO: end of the temporary code that will be removed.
+
+    try {
+      await this.assessmentService.studentConfirmAssessment(
+        originalAssessment.id,
+        student.id,
+      );
+    } catch (error) {
+      switch (error.name) {
+        case ASSESSMENT_NOT_FOUND:
+          throw new NotFoundException(error.message);
+        case ASSESSMENT_INVALID_OPERATION_IN_THE_CURRENT_STATE:
+          throw new UnprocessableEntityException(error.message);
```

Comment:
are we planning to add @apitags for these errors, or we are planning to do it all in a future ticket? just a qn

Reply:
This controller will be moved and changed so I avoided adding the tags here.